regression-test/suites/query_p0/sql_functions/array_functions/test_array_functions_by_literal.groovy

```diff
@@ -219,6 +219,12 @@ suite("test_array_functions_by_literal") {
     qt_sql "select (array(cast ('2023-02-06' as datev2), cast ('2023-02-05' as datev2)))[1:2]"
     qt_sql "select (array(cast (111.111 as decimalv3(6,3)),cast (222.222 as decimalv3(6,3))))[1:2]"
 
+    // array_intersect
+    qt_sql_intersect_1 "select array_intersect([1,2,3], [1,2,3], [null])"
```

Comment:
Is the result stable? For example, sometimes return [null, 1] , sometimes return [1, null] ？